flink-streaming-connectors/flink-connector-cassandra/src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java

```diff
@@ -389,4 +403,54 @@ private TestCassandraTupleWriteAheadSink(String tableName, TypeSerializer<IN> se
 			this.tableName = tableName;
 		}
 	}
+
+	@Test
+	public void testCassandraScalaTupleAtLeastOnceSinkBuilderDetection() throws Exception {
+		Class<Tuple1<String>> c = (Class<Tuple1<String>>) new Tuple1<>("hello").getClass();
+		Seq<TypeInformation<?>> typeInfos = JavaConverters.asScalaBufferConverter(
+			Arrays.<TypeInformation<?>>asList(new TypeInformation[]{BasicTypeInfo.STRING_TYPE_INFO})).asScala();
+		Seq<String> fieldNames = JavaConverters.asScalaBufferConverter(
+			Arrays.asList(new String[]{"_1"})).asScala();
+
+		CaseClassTypeInfo<Tuple1<String>> typeInfo = new CaseClassTypeInfo<Tuple1<String>>(c, null, typeInfos, fieldNames) {
+			@Override
+			public TypeSerializer<Tuple1<String>> createSerializer(ExecutionConfig config) {
+				return null;
+			}
+		};
+
+		StreamExecutionEnvironment env = mock(StreamExecutionEnvironment.class);
```

Comment:
I think there is no need to mock this. You can use a regular execution environment.
Unless you call `execute()`, no mini cluster would be spawned anyways.